runtime/oti/jvminit.h

```diff
@@ -253,6 +253,8 @@ enum INIT_STAGE {
 #define VMOPT_XFASTRESOLVE "-Xfastresolve"
 #define VMOPT_XSHARECLASSES "-Xshareclasses"
 #define VMOPT_XSHARECLASSES_COLON "-Xshareclasses:"
+#define VMOPT_XSAVERAMSTATE "-Xsaveramstate"
```

Comment:
Typically, if we expect an equals we add the equals to the search string. This will let you do different things if `-Xsaveramstate` is specified vs `-Xsaveramstate=/some/dir`

E.g. 

`-Xshareclasses` -> create a cache with the default name in the current dir

 `-Xshareclasses:name=MyCache` -> create a cache called MyCache in current dir

Now, you don't need to provide a default behaviour for `-Xsaveramstate`, but since you always expect the option to have an argument, a `-Xsaveramstate=` search string would be better.

Reply:
GET_OPTIONS_VAL takes in a delim character to find where to substr after.

I could define the macro as -Xsaverramstate:file, and then still pass '=' as the delim character.

Reply:
> GET_OPTIONS_VAL takes in a delim character to find where to substr after.

Yes, but this is done after a `FIND_AND_CONSUME..`. What I was talking about earlier is the search string for the `FIND_AND_CONSUME..`. If you look at other examples where the -Xoption is delimited with an `=` or a `:`, (eg. VMOPT_XSERVICE_EQUALS, VMOPT_XDUMP_DIRECTORY_EQUALS, etc.). The search string will always include the delimiter.

This future proofs the arg parsing in case we decide the add an option with the same name and does not contain the delimeter (eg. -Xshareclasses and -Xshareclasses:). Or, if the option name happens to be a substring of another option.

Reply:
In any case this is a minor point, it shouldn't impede your progress.

Reply:
ah I see what you mean. Fixed it